lib/parser.js

```diff
@@ -129,7 +132,68 @@ function createValuesObject(valuesFilePath) {
   return resultValues;
 }
 
+/*
+ * Returns the array of Parameters after appending the cart dependencies
+ * The function reads Chart.yaml file, goes over dependencies array,
+ * sends axios GET request to the repository of the dependency chart, reads the home page,
+ * and injects home page Parameter by finding any first dependency chart variable
+ */
+async function appendDependencies(dependencyPath, valuesMetadata) {
+  const chart = YAML.parse(fs.readFileSync(dependencyPath, 'utf8'));
+
+  axiosRetry(axios, {
+    retries: 3,
+    retryDelay: (retryCount) => {
+      console.log(`INFO: retrying to connect to url, attempt ${retryCount}...`);
+      return 0;
+    }
+  });
+
+  let requests = [];
+
+  console.log(`INFO: receiving dependencies home pages...`);
+  for (const subChart of chart.dependencies) {
+    const repositoryUrl = subChart.repository + "/index.yaml";
+
+    // skip if its already defined
+    if (valuesMetadata.find(param => param.name === subChart.name)) {
+      continue;
+    }
+
+    // example: subChart.name == minio then search for minio.something
+    const firstParam = valuesMetadata.find(param => {
+      paramStartName = param.name.match(/([^.]+)[.]/);
+      return paramStartName && paramStartName[1] === subChart.name;
+    })
+
+    // skip if didnt found first subchart description
+    if (!firstParam) {
+      continue;
+    }
+
+    const index = valuesMetadata.indexOf(firstParam);
+
+    requests.push(axios.get(repositoryUrl).then((response) => {
+      if (response.status !== 200) {
+        throw new Error(`ERROR: couldnt connection failed to "${repositoryUrl}", return status code: ${err.response.status} after 3 retry attempts`);
+      } else {
+        const param =  new Parameter(subChart.name);
+        const homeUrl = YAML.parse(response.data).entries[subChart.name][0].home;
+        param.description = `For additional variables configurations please refer [here](${homeUrl})`;
+        param.validate = false;
+
+        param.section = firstParam.section;
+        valuesMetadata.splice(index, 0, param);
+      }
+    }));
+  }
+
+  await Promise.all(requests);
+  console.log(`INFO: successfully received dependencies home pages!`);
+}
+
```

Comment:
Note we can have the subcharts are defined into the `Chart.yaml` and the sub-charts exist in the `charts` directory just by executing `helm dependencies update`. 
IMHO querying the chart repository looks overcomplicating things for what we are trying to achieve here.

Reply:
By default the values overridden from the default dependency parameters are shown in the README since you need to override them into the parent `values.yaml`. 
This PR if I am not wrong is adding the whole list of parameters for the dependencies. I would try to avoid sending manual requests when it is possible to use built-in commands. Another possible is approach is to rely on the user to run `helm dependencies update` before running the tool.

Reply:
I pretty much agree with you that its better to use built-in commands instead of doing my own requests.

I guess I can do something like when he usses the "dependencies" flag then i will check if there are the subcharts, and it not then alert "please run `helm dependencies update` before using the flag"

what about the destination, im still in a big dilemma if I should update the values.yaml, or the final README.md:
- should I do something like:
  `readme-generator --readme README.md --values values.yaml --dependency-dir ./`
  and it will inject those variables into README.md
- or should i do something like:
  `readme-generator --values values.yaml --dependency-dir ./`
  and it will inject those variables into values.yaml (with `@extra home page blah blah...`)

Reply:
>I guess I can do something like when he usses the "dependencies" flag then i will check if there are the subcharts, and it not then alert "please run helm dependencies update before using the flag"

+1 to this. I think that is the correct approach.

About the second part, do not change the `values.yaml`. We consider the `values.yaml` as the source of truth so it should not be changed automatically.
What I would do is to add a boolean flag for dependencies (`---include-dependencies`) and if it is provided include the dependencies into the `README.md`. Not you don't need the dependencies path since the Helm Chart structure is always the same so it should be at the same level as the `values.yaml` but into the `charts` folder.

Reply:
> Not you don't need the dependencies path since the Helm Chart structure is always the same so it should be at the same level as the values.yaml but into the charts folder.

true, i just saw that there is path for each file so i thought we separate everything.
but yes its really for the best to just relay on values.yaml path.

i will probably do it on the weeken when i will get free

Reply:
hello,
previously with axios request it was taking around 8 seconds to detect the home pages of the subcharts.
now it takes me around 2 minutes (im not even joking) only to run the command `helm dependencies update`.
(i was testing the chart `bitnami/thanos`, and it has only 2 subcharts, im afraid to think of charts that have more dependencies)

it makes everything super slow, im not sure if its worth it..

Reply:
what about developing other tool, that will send the axios request and modify the values.yaml?

this way the readme-generator will always stay fast, because he will have everything prepared in the values.yaml (with the `@extra home page...` marks),

and the other tool will resolve everything in only 5-10 seconds instead of 1-5 minutes, and just add the `@extra home page...` in the values.yaml.